netket/graph/symmetry.py

```diff
@@ -105,20 +105,16 @@ def __inverse(self):
 
         return inverse
 
-    def __product_table(self, inverse):
+    def __product_table(self):
         """
         Returns a product table over the group where the columns use the involution
         of the group. If :code:`g = self[self.inverse()[element]]', :code:`h = self[element2]`
         and code:`u = self[product_table()[element,element2]], we are
         solving the equation u = gh
-
-        Arguments:
-            inverse: The indices of the group involution as returned by self.__inverse()
         """
 
         automorphisms = self.to_array()
-        inverse = automorphisms[inverse].squeeze()
-
+        inverse = automorphisms[self.__inverse()].squeeze()
```

Comment:
This should be `self.inverse()`, though, so that `__inverse` is really only called when no cached result is available.
```suggestion
        inverse = automorphisms[self.inverse()].squeeze()
```

Reply:
ohh yeah my mistake